acrg/BC/BC_CAMS.py

```diff
@@ -54,70 +54,108 @@
 
 acrg_path = Paths.acrg
 data_path = Paths.data
-cams_directory = os.path.join(data_path, 'ECMWF_CAMS', 'CAMS_inversion/')
+cams_directory_default = os.path.join(data_path, 'ECMWF_CAMS', 'CAMS_inversion/')
+
+# default variable names in the CAMS files for each species and CAMS version, add 
+default_inputs = {'ch4': {'v19': {'altitude': 'altitude',
+                                  'file_start_str': 'cams73',
+                                  'height': 'height',
+                                  'hlevel': 'hlevel',
+                                  'lat': 'latitude',
+                                  'level': 'level',
+                                  'lon': 'longitude',
+                                  'time': 'time',
+                                  'z': 'z'}
+                         }
+                   }
+
+default_inputs['ch4']['latest'] = default_inputs['ch4']['v19']
 
 with open(os.path.join(acrg_path, "data/species_info.json")) as f:
     species_info=json.load(f,object_pairs_hook=OrderedDict)
 
-def readCAMSInversion(start, end, species='ch4', cams_directory=cams_directory,
-                      version='v19'):
+def readCAMSInversion(start, end, species='ch4', cams_directory=cams_directory_default,
+                      version='v19', variables=None):
     '''
     Args:
         start, end (str)
             Start and end dates
-        species (str)
-        cams_directory (str)
+        species (str, optional)
+            defaults to 'ch4'
+        cams_directory (str, optional)
             Path to CAMS files
-        version (str, int)
+        version (str, int, optional))
             version number of inversion, e.g. 'v19'
             if an int is entered, it will be converted to a str starting with 'v'
+            defaults to 'v19'
+        variables (dict, optional)
+            dictionary containing variable names including:
+                'altitude', 'file_start_str', 'height', 'hlevel', 'lat', 'lon', 'time', & 'z'
+            if not given variables will be assumed to be the same as for CAMS CH4 v19
             
     Returns:
         xarray dataset of combined CAMS inversions
     '''
+    # set default inputs if not given
+    cams_directory = cams_directory_default if cams_directory is None else cams_directory
+    version = 'v19' if version is None else version
```

Comment:
Are these default definitions still needed?

And could the default be set to "latest" now? That could be nicely self-documenting.